From 378974ae0a44d6cf5468e43db23f576e5ca81bff Mon Sep 17 00:00:00 2001
From: Venkat Jayaraman <venkat.jayaraman@intel.com>
Date: Thu, 6 Sep 2012 16:50:20 -0700
Subject: [PATCH 24/78] WLAN: Skip bind/unbind when no Wi-Fi module present

BZ: 55966

unbind/bind sequence in the case of updating only MAC address without
calibration, does not check for presence of Wi-Fi module

Change-Id: I9e85a1b0e2954cb68600df4b1260e98936be83ab
Signed-off-by: Prabu Renganathan <prabu.renganathan@intel.com>
Signed-off-by: Bin Gao <bin.gao@intel.com>
Signed-off-by: Venkat Jayaraman <venkat.jayaraman@intel.com>
Reviewed-on: http://android.intel.com:8080/64718
Reviewed-by: Tricot, HonoreX <honorex.tricot@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77981
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wlan_prov/wlan_provisioning.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/wlan_prov/wlan_provisioning.c b/wlan_prov/wlan_provisioning.c
index a79513b..9531497 100644
--- a/wlan_prov/wlan_provisioning.c
+++ b/wlan_prov/wlan_provisioning.c
@@ -245,6 +245,11 @@ int main(int argc, char **argv)
 		return -1;
 	}
 
+       if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)) {
+		LOGE("no wlan device detected, exit...");
+		return -1;
+       }
+
 #ifdef BUILD_WITH_CHAABI_SUPPORT
 	/* Read MAC address from Chaabi */
 	if (get_customer_data(ACD_WLAN_MAC_ADDR_FIELD_INDEX,
@@ -323,8 +328,7 @@ end:
 		* account. In that case, the reboot is required after flashing for the
 		* time the board.
 		*/
-		if (sdio_get_pci_id(SYSFS_SDIO_DEVICES_PATH, device_id)
-			|| unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
+		if (unbind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)
 			|| bind_wlan_sdio_drv(WLAN_SDIO_BUS_PATH, device_id)) {
 			/*
 			* Rebooting the board: In this level, the NVS was saved. The
-- 
1.7.9

